src/Monitor/Tracing/Zipkin.hs

```diff
@@ -296,14 +297,21 @@ insertTag key val =
   Endo $ \bldr -> bldr { builderTags = Map.insert key (JSON.toJSON val) (builderTags bldr) }
 
 importB3 :: B3 -> Endo Builder
-importB3 b3 =
+importB3 b3 = importB3traceAndPolicy b3 <> 
+              Endo (\bldr -> bldr { builderSpanID = Just (b3SpanID b3) })
+
+importB3asParent :: B3 -> Endo Builder
+importB3asParent b3 = importB3traceAndPolicy b3 <> 
+                      Endo (\bldr -> bldr { builderReferences = Set.singleton (ChildOf $ b3SpanID b3) })
+
+importB3traceAndPolicy :: B3 -> Endo Builder
+importB3traceAndPolicy b3 =
```

Comment:
With the two functions above renamed, let's simplify this one's name back to `importB3`.